stdlib/format.mli

```diff
@@ -508,16 +505,23 @@ val safe_set_geometry : max_indent:int -> margin:int -> unit
    [pp_set_geometry ppf ~max_indent ~margin] sets both the margin
    and maximum indentation limit for [ppf].
 
+   [margin] and [max_indent] should satisfy
+   [1 < max_indent < margin < 1,000,000,010].
+
+   If the upper limit of [1,000,000,0010] is exceeded, even if the inequality
+   [max_indent < margin] is satisfied, the [max_indent] will not be set
+   correctly.
+
+   If [max_indent >= margin], [pp_set_geometry] raises an invalid argument
+   exception whereas [pp_safe_set_geometry] does nothing.
```

Comment:
Overall, this text is  focusing too much on a corner case. I would suggest to rather amend the description of the domain of `pp_set_geometry` below by exposing `pp_infinity` in the interface and make `pp_set_geometry` raises an exception uniformly outside of this domain.

Reply:
Do you mean something like the latest commit? I am new to `ocaml` so any guidance would be appreciated. I made the following changes:

1) Added `val pp_inifinty : int` to `format.mli`. I added a line documentating it, which I paraphrased from the documentation in `format.ml`. Is this sufficient? 

2) Added a check for `margin >= pp_infinity` after `margin <= max_indent` in `validate_geometry`. This way, the bound on `max_indent` is automatically satisfied.

3) Updated the documentation for `check_geometry`, since it also now checks for the upper bound. I thought that this would make things consistent.

4) Updated the documentation for `pp_set_geometry` by simply adding the upper bound as `pp_infinity`. The rest was reverted back to what it was in the current `trunk`.

As a final question, is there somewhere I should write tests for this code?